From 56ee9c989b83ccd7de657cbcefc73e4f3f404951 Mon Sep 17 00:00:00 2001
From: SangeethaRao <sangeetha.rao@intel.com>
Date: Tue, 14 Apr 2015 23:23:49 -0500
Subject: [PATCH 35/59] arch/arm/mach-axxia: Updated Axxia 55xx PCIe driver to
 use correct Doorbell interrupt IRQs

git://git.yoctoproject.org/linux-yocto-4.1 standard/axxia/base
	commit f2b726bce878081cdfacd84d434e9293ebe73dd5

Signed-off-by: SangeethaRao <sangeetha.rao@intel.com>
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm/mach-axxia/pci.c |   15 +++++++--------
 1 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/arch/arm/mach-axxia/pci.c b/arch/arm/mach-axxia/pci.c
index a734cad..bec29e1 100644
--- a/arch/arm/mach-axxia/pci.c
+++ b/arch/arm/mach-axxia/pci.c
@@ -662,14 +662,6 @@ static int axxia_pcie_setup(int portno, struct pci_sys_data *sys)
 		goto fail;
 	}
 
-	/* MSI interrupts */
-	for (i = 1; i <= 16; i++) {
-		port->irq[i] = irq_of_parse_and_map(port->node, i);
-		if (!port->irq[i])
-			break;
-		irq_set_chained_handler(port->irq[i], pcie_msi_irq_handler);
-	}
-
 	/* Setup as root complex */
 	pci_config = readl(port->regs + PCIE_CONFIG);
 	pci_status = readl(port->regs + PCIE_STATUS);
@@ -730,6 +722,13 @@ static int axxia_pcie_setup(int portno, struct pci_sys_data *sys)
 		}
 	}
 
+	/* MSI interrupts */
+	for (i = 2; i <= 17; i++) {
+		port->irq[i] = irq_of_parse_and_map(port->node, i);
+		if (!port->irq[i])
+			break;
+		irq_set_chained_handler(port->irq[i], pcie_msi_irq_handler);
+	}
 	/*
 	 * Setup outbound PCI Memory Window
 	 */
-- 
1.7.5.4

